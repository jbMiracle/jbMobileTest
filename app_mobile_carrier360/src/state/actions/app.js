import { App as AppActionTypes } from '../types';

export const genesis = () => ({
  type: AppActionTypes.Genesis,
});

export const forgotPassword = email => ({
  type: AppActionTypes.ForgotPassword,
  payload: email,
});

export const forgotPasswordServiceDone = (success, errorMessage) => ({
  type: AppActionTypes.ForgotPasswordDone,
  payload: {
    success,
    errorMessage,
  },
});
