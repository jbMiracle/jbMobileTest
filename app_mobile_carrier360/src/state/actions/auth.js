import { Auth as AuthActionTypes } from '../types';

export const userAuthenticated = (result = true) => ({
  type: AuthActionTypes.UserAuthenticated,
  payload: result,
});

export const queryBiometricSupport = () => ({
  type: AuthActionTypes.QueryBiometricSupport,
});

export const biometricSupportDetermined = supportResult => ({
  type: AuthActionTypes.BiometricSupportDetermined,
  payload: supportResult,
});

export const setupBiometricLogin = () => ({
  type: AuthActionTypes.SetupBiometricLogin,
});

export const biometricLoginSetup = (isSetup = true) => ({
  type: AuthActionTypes.BiometricLoginSetup,
  payload: { isSetup },
});
