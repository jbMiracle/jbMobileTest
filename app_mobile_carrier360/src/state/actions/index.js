import * as App from './app';
import * as Auth from './auth';
import * as UserAccess from './userAccess/UserAccessActions';

export {
  App,
  Auth,
  UserAccess,
};
