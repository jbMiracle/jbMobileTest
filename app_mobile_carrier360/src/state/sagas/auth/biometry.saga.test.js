import SagaTester from 'redux-saga-tester';
import { Auth as AuthActions } from '../../actions';
import authSaga from '.';

describe('BiometrySaga', () => {
  let sagaTester = null;
  beforeEach(() => {
    sagaTester = new SagaTester({});
    sagaTester.start(authSaga);
  });

  test('queryBiometricSupport', async () => {
    sagaTester.dispatch(AuthActions.queryBiometricSupport());
    expect(sagaTester.getState()).toMatchSnapshot();
  });
  test('biometricSupportDetermined', async () => {
    sagaTester.dispatch(AuthActions.biometricSupportDetermined('face_id'));
    expect(sagaTester.getState()).toMatchSnapshot();
  });
  test('setupBiometricLogin', async () => {
    sagaTester.dispatch(AuthActions.setupBiometricLogin());
    expect(sagaTester.getState()).toMatchSnapshot();
  });
  test('biometricLoginSetup', async () => {
    sagaTester.dispatch(AuthActions.biometricLoginSetup());
    expect(sagaTester.getState()).toMatchSnapshot();
  });
  test('userAuthenticated', async () => {
    sagaTester.dispatch(AuthActions.userAuthenticated());
    expect(sagaTester.getState()).toMatchSnapshot();
  });
});
