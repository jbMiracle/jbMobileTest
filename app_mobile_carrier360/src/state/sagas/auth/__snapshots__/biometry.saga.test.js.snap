// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BiometrySaga biometricLoginSetup 1`] = `Object {}`;

exports[`BiometrySaga biometricSupportDetermined 1`] = `Object {}`;

exports[`BiometrySaga queryBiometricSupport 1`] = `Object {}`;

exports[`BiometrySaga setupBiometricLogin 1`] = `Object {}`;

exports[`BiometrySaga userAuthenticated 1`] = `Object {}`;
